lib/jsdom/browser/Window.js

```diff
@@ -182,24 +188,56 @@ function Window(options) {
   ///// METHODS
 
   let latestTimerId = 0;
+  let latestAnimationFrameCallbackId = 0;
 
   this.setTimeout = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, timers, args);
   };
   this.setInterval = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, timers, args);
+  };
+  this.clearInterval = stopTimer.bind(this, timers);
+  this.clearTimeout = stopTimer.bind(this, timers);
+
+  if (this._pretendToBeVisual) {
+    this.requestAnimationFrame = fn => {
+      const hr = hrtime(windowInitialized);
+      const hrInNano = hr[0] * 1e9 + hr[1];
+      const fps = 1000 / 60;
+      const hrInMicro = hrInNano / 1e6;
+
+      return startTimer(
+        window,
+        setTimeout,
+        clearTimeout,
+        ++latestAnimationFrameCallbackId,
+        fn,
+        fps,
```

Comment:
It occurs to me that while in a browser if you did something like:

rAF
wait 6 milliseconds
rAF
wait 6 milliseconds
rAF

then you'd probably get all callbacks called together, instead of at now + 16 ms, now + 22 ms, now + 28 ms. But, we can leave it like this, and see if anyone complains...

Reply:
I've tried to test this in Chrome, but at least using `setTimeout` I'm unable to have them fire at once.

```js
requestAnimationFrame((t)=>{console.log('hello', t, performance.now())});
setTimeout(() => requestAnimationFrame((t)=>{console.log('hello delayed', t, performance.now())}), 1);
```

Prints e.g.

```
hello 2098569.228999957 2098571.895
hello delayed 2098585.90299997 2098587.875
```

Running them multiple times never have them fire at once